jenkinsci/tuleap-api-plugin
# Add all the authentication queries in the plugin.

This is a part of [story #14018](https://tuleap.net/plugins/tracker/?aid=14018) have a central management of users and groups using oauth

You can test this commit with the (future) tuleap-oauth plugin (not released yet):
 - Pull "test/tuleap-api-integration branch" from tuleap-oauth plugin. (The master branch is not compatible with this commit yet)
 - You have to install this plugin manually :
  - In this plugin, execute: mvn clean install (or mvn package)
  - Go to the tuleap-oauth plugin directory
  - mvn install:install-file -DgroupId=io.jenkins.plugins -DartifactId=tuleap-api -Dversion=1.0.3-SNAPSHOT -Dfile=path/of/your/tuleap-api.hpi_file -Dpackaging=hpi
  - mvn install:install-file -DgroupId=io.jenkins.plugins -DartifactId=tuleap-api -Dversion=1.0.3-SNAPSHOT -Dfile=path/of/your/tuleap-api.jar_file
  - mvn clean install (Test if all are ok)
 - mvn hpi:run
 - Go to the Global Configuration page to configure your Tuleap instance URL
 - Go to the Security Realm page and configure the Tuleap Authentication.
=> You should be able to be logged in Jenkins with the used user to be logged in Tuleap

Have fun :)
